### ISSUE TITLE
New rule: no-useless-continue

### ISSUE THREAD

**Issue by Fabrice-TIERCELIN** [eslint/eslint#13608]
**Please describe what the rule should do:**
Removes the continue statement that provides nothing to the code behavior.

**What new ECMAScript feature does this rule relate to?**

None.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
var i = 0;
while (i < 10) {
  i++;
  continue;
}
```

```js
for (i = 0; i < 10; i++) {
  continue;
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
It is universal and useful for any EcmaScript code. `no-useless-continue` should be included in ESLint as long as `no-useless-return` is included.

**Are you willing to submit a pull request to implement this rule?**
#13607

**Comment by mdjermanovic** [comment]
Hi @Fabrice-TIERCELIN, thanks for the new rule proposal!

I think this is a valid rule, but we have decided to not accept any new "suggestion" rules if they are not related to new ECMAScript features (please find more details in [this blog post](https://eslint.org/blog/2020/05/changes-to-rules-policies)).

We can always consider adding new rules that catch possible errors, even if it isn't related to new features, but this looks more like a suggestion type rule.

**Comment by Fabrice-TIERCELIN** [comment]
OK.

In which project do you suggest me to add this code? Any plugin?

**Comment by mdjermanovic** [comment]
> OK.
> 
> In which project do you suggest me to add this code? Any plugin?

You can search for `eslint-plugin` on npm, or find some plugins here:

https://github.com/dustinspecker/awesome-eslint

I'd guess this rule is probably best fit for general-purpose code quality plugins, like `eslint-plugin-unicorn`.
